### ISSUE TITLE
HTTP backend requests & responses are not logged the log TRACE level is enabled

### ISSUE THREAD

**Issue by ag-TJNII** [opentofu/opentofu#2003]
### OpenTofu Version

```shell
OpenTofu v1.8.2
on linux_amd64
```

### OpenTofu Configuration Files

```hcl
terraform {
  backend "http" {
    address = "https://tf-state-server.[domain]/api/v0/tf/backend/state/[resource-id]"
    lock_address = "https://tf-state-server.[domain]/api/v0/tf/backend/lock/[resource-id]"
    lock_method = "LOCK"
    unlock_address = "https://tf-state-server.[domain]/api/v0/tf/backend/unlock/[resource-id]"
    unlock_method = "UNLOCK"
  }
}
```

### Debug Output

```
2024-09-20T15:20:19.766Z [INFO]  OpenTofu version: 1.8.2
2024-09-20T15:20:19.767Z [DEBUG] using github.com/hashicorp/go-tfe v1.36.0
2024-09-20T15:20:19.767Z [DEBUG] using github.com/opentofu/hcl/v2 v2.0.0-20240416130056-03228b26f391
2024-09-20T15:20:19.767Z [DEBUG] using github.com/hashicorp/terraform-svchost v0.1.1
2024-09-20T15:20:19.767Z [DEBUG] using github.com/zclconf/go-cty v1.14.4
2024-09-20T15:20:19.767Z [INFO]  Go runtime version: go1.22.6
2024-09-20T15:20:19.767Z [INFO]  CLI args: []string{"/usr/local/bin/terraform", "plan", "-var-file=[SANITIZED].tfvars"}
2024-09-20T15:20:19.767Z [TRACE] Stdout is not a terminal
2024-09-20T15:20:19.767Z [TRACE] Stderr is not a terminal
2024-09-20T15:20:19.767Z [TRACE] Stdin is not a terminal
2024-09-20T15:20:19.767Z [DEBUG] Attempting to open CLI config file: /home/tfuser/.tofurc
2024-09-20T15:20:19.767Z [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2024-09-20T15:20:19.767Z [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
2024-09-20T15:20:19.767Z [DEBUG] ignoring non-existing provider search directory /home/tfuser/.terraform.d/plugins
2024-09-20T15:20:19.767Z [DEBUG] ignoring non-existing provider search directory /home/tfuser/.local/share/terraform/plugins
2024-09-20T15:20:19.767Z [DEBUG] ignoring non-existing provider search directory /usr/local/share/terraform/plugins
2024-09-20T15:20:19.767Z [DEBUG] ignoring non-existing provider search directory /usr/share/terraform/plugins
2024-09-20T15:20:19.768Z [DEBUG] Found the config directory: /home/tfuser/.terraform.d
2024-09-20T15:20:19.768Z [INFO]  CLI command args: []string{"plan", "-var-file=[SANITIZED].tfvars"}
2024-09-20T15:20:19.778Z [TRACE] Meta.Backend: built configuration for "http" backend with hash value 1003972498
2024-09-20T15:20:19.778Z [TRACE] Preserving existing state lineage "1760ced6-473a-43a6-4cab-b3cbf8a27439"
2024-09-20T15:20:19.779Z [TRACE] Preserving existing state lineage "1760ced6-473a-43a6-4cab-b3cbf8a27439"
2024-09-20T15:20:19.780Z [TRACE] Meta.Backend: working directory was previously initialized for "http" backend
2024-09-20T15:20:19.780Z [TRACE] Meta.Backend: using already-initialized, unchanged "http" backend configuration
2024-09-20T15:20:19.782Z [TRACE] Meta.Backend: instantiated backend of type *http.Backend
2024-09-20T15:20:19.784Z [TRACE] providercache.fillMetaCache: scanning directory .terraform/providers
2024-09-20T15:20:19.784Z [TRACE] getproviders.SearchLocalDirectory: found registry.opentofu.org/ag-tjnii/powerdns v101.6.1 for linux_amd64 at .terraform/providers/registry.opentofu.org/ag-tjnii/powerdns/101.6.1/linux_amd64
2024-09-20T15:20:19.785Z [TRACE] getproviders.SearchLocalDirectory: found registry.opentofu.org/hashicorp/archive v2.6.0 for linux_amd64 at .terraform/providers/registry.opentofu.org/hashicorp/archive/2.6.0/linux_amd64
2024-09-20T15:20:19.785Z [TRACE] getproviders.SearchLocalDirectory: found registry.opentofu.org/hashicorp/aws v5.66.0 for linux_amd64 at .terraform/providers/registry.opentofu.org/hashicorp/aws/5.66.0/linux_amd64
2024-09-20T15:20:19.786Z [TRACE] getproviders.SearchLocalDirectory: found registry.opentofu.org/hashicorp/external v2.3.4 for linux_amd64 at .terraform/providers/registry.opentofu.org/hashicorp/external/2.3.4/linux_amd64
2024-09-20T15:20:19.786Z [TRACE] getproviders.SearchLocalDirectory: found registry.opentofu.org/hashicorp/local v2.5.1 for linux_amd64 at .terraform/providers/registry.opentofu.org/hashicorp/local/2.5.1/linux_amd64
2024-09-20T15:20:19.786Z [TRACE] getproviders.SearchLocalDirectory: found registry.opentofu.org/hashicorp/null v3.2.2 for linux_amd64 at .terraform/providers/registry.opentofu.org/hashicorp/null/3.2.2/linux_amd64
2024-09-20T15:20:19.786Z [TRACE] getproviders.SearchLocalDirectory: found registry.opentofu.org/hashicorp/vault v2.24.1 for linux_amd64 at .terraform/providers/registry.opentofu.org/hashicorp/vault/2.24.1/linux_amd64
2024-09-20T15:20:19.786Z [TRACE] providercache.fillMetaCache: including .terraform/providers/registry.opentofu.org/hashicorp/null/3.2.2/linux_amd64 as a candidate package for registry.opentofu.org/hashicorp/null 3.2.2
2024-09-20T15:20:19.787Z [TRACE] providercache.fillMetaCache: including .terraform/providers/registry.opentofu.org/hashicorp/vault/2.24.1/linux_amd64 as a candidate package for registry.opentofu.org/hashicorp/vault 2.24.1
2024-09-20T15:20:19.787Z [TRACE] providercache.fillMetaCache: including .terraform/providers/registry.opentofu.org/ag-tjnii/powerdns/101.6.1/linux_amd64 as a candidate package for registry.opentofu.org/ag-tjnii/powerdns 101.6.1
2024-09-20T15:20:19.787Z [TRACE] providercache.fillMetaCache: including .terraform/providers/registry.opentofu.org/hashicorp/archive/2.6.0/linux_amd64 as a candidate package for registry.opentofu.org/hashicorp/archive 2.6.0
2024-09-20T15:20:19.787Z [TRACE] providercache.fillMetaCache: including .terraform/providers/registry.opentofu.org/hashicorp/aws/5.66.0/linux_amd64 as a candidate package for registry.opentofu.org/hashicorp/aws 5.66.0
2024-09-20T15:20:19.787Z [TRACE] providercache.fillMetaCache: including .terraform/providers/registry.opentofu.org/hashicorp/external/2.3.4/linux_amd64 as a candidate package for registry.opentofu.org/hashicorp/external 2.3.4
2024-09-20T15:20:19.787Z [TRACE] providercache.fillMetaCache: including .terraform/providers/registry.opentofu.org/hashicorp/local/2.5.1/linux_amd64 as a candidate package for registry.opentofu.org/hashicorp/local 2.5.1
2024-09-20T15:20:21.280Z [TRACE] providercache.fillMetaCache: using cached result from previous scan of .terraform/providers
2024-09-20T15:20:21.315Z [TRACE] providercache.fillMetaCache: using cached result from previous scan of .terraform/providers
2024-09-20T15:20:21.355Z [TRACE] providercache.fillMetaCache: using cached result from previous scan of .terraform/providers
2024-09-20T15:20:21.397Z [TRACE] providercache.fillMetaCache: using cached result from previous scan of .terraform/providers
2024-09-20T15:20:21.467Z [TRACE] providercache.fillMetaCache: using cached result from previous scan of .terraform/providers
2024-09-20T15:20:21.533Z [TRACE] providercache.fillMetaCache: using cached result from previous scan of .terraform/providers
2024-09-20T15:20:21.577Z [DEBUG] checking for provisioner in "."
2024-09-20T15:20:21.578Z [DEBUG] checking for provisioner in "/usr/bin"
2024-09-20T15:20:21.578Z [TRACE] Meta.Backend: backend *http.Backend does not support operations, so wrapping it in a local backend
2024-09-20T15:20:21.582Z [INFO]  backend/local: starting Plan operation
2024-09-20T15:20:21.582Z [TRACE] backend/local: requesting state manager for workspace "default"
2024-09-20T15:20:21.582Z [TRACE] backend/local: requesting state lock for workspace "default"
2024-09-20T15:20:21.582Z [DEBUG] LOCK https://tf-state-server.[domain]/api/v0/tf/backend/lock/[resource-id]
Acquiring state lock. This may take a few moments...
╷
│ Error: Error acquiring the state lock
│
│ Error message: HTTP remote state already locked: 
│ ID=bf014658-40d1-1a81-bd3d-2d9381a0db4a
│ Lock Info:
│   ID:        14064fbf-01a4-f5ab-bb41-1327f9decab4
│   Path:
│   Operation: OperationTypePlan
│   Who:       tfuser@70ccfbe17e74
│   Version:   1.8.2
│   Created:   2024-09-20 15:20:21.582465339 +0000 UTC
│   Info:
│
│
│ OpenTofu acquires a state lock to protect the state from being written
│ by multiple users at the same time. Please resolve the issue above and try
│ again. For most commands, you can disable locking with the "-lock=false"
│ flag, but this is not recommended.
╵
```

### Expected Behavior

I'm debugging why the lock info shown is not correct, trying to determine if it is a backend server bug or a Tofu bug.  I would have expected to see the details of the HTTP lock request & server response at trace level.

### Actual Behavior

Only a `LOCK [URL]` line was logged.  No request or response details.

### Steps to Reproduce

- Configure Tofu with a http backend
- Perform an operation at trace log level

### Additional Context

There's an argument to not log the actual state payloads as that it likely sensitive and may be large.  For that it would be nice to at least log the response headers.  Locks are not sensitive so logging the lock payloads should not be a security concern and would be useful for debugging.

**Comment by ag-TJNII** [comment]
Upon further thought I believe this would be better tagged as a feature request.  As a random person from the internet I don't appear to have the permissions to do this.  I will not be upset if this is downgraded from a bug to a feature request.

**Comment by haiderbari** [comment]
Hey core team can someone assign the issue to me I am happy to take it on. @ag-TJNII is also happy for me to take it on. Thanks again @ag-TJNII for opening the issue letting me take it on.

**Comment by haiderbari** [comment]
Hey I am currently working on this, just wanted to know if logging the header is necessary. If me logging it is unnecessary as really don't return any info other than content type and status code. I think CLI  doesn't use the headers so i feel it's unnecessary unless people have valid reason for logging headers. Currently I am logging the status code and the post body for non 200 status codes

**Comment by haiderbari** [comment]
Hi just want to clarify something do you want to log request headers as well with response headers. I can't see why you want to return authorization with the response headers. I can add blacklist of masking certain values but I can't see what sensitive data would come back from response headers only request headers. If you want the request headers log no problem

**Comment by haiderbari** [comment]
Hey @abstractionfactory I was thinking about that as well. Are we happy to only log the response headers and mask cookie details

**Comment by ollevche** [comment]
Resolved in #2120
